google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/MaterializedViewDefinition.java

```diff
@@ -149,6 +197,28 @@ static MaterializedViewDefinition fromPb(Table tablePb) {
       if (materializedViewDefinition.getRefreshIntervalMs() != null) {
         builder.setRefreshIntervalMs(materializedViewDefinition.getRefreshIntervalMs());
       }
+      if (tablePb.getTimePartitioning() != null) {
+        try {
+          builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+        } catch (IllegalArgumentException e) {
+          throw new IllegalArgumentException(
+              "Illegal Argument - Got unexpected time partitioning "
+                  + tablePb.getTimePartitioning().getType()
```

Comment:
Do you want to intercept like this?  Seems like we may want the other fromPb() message functions to be responsible for throwing a human readable error?

Reply:
Okay - I removed this interception.